### ISSUE TITLE
[LineAnnotation] improve accessibility (a11y)

### ISSUE THREAD

**Issue by alexwizp** [elastic/elastic-charts#2520]
This issue is related to [https://github.com/elastic/kibana/issues/188861](https://github.com/elastic/kibana/issues/188861). During an a11y audit, we found a couple of issues with that component that we should address.

1. **All modes (clickable and icon modes)**: The annotation tooltip should be keyboard accessible. Currently, when rendered as a button, focus can be set using the keyboard, but the tooltip does not open. In icon mode, keyboard navigation does not work at all. This violates the [WCAG 2.2 guidelines for keyboard accessibility](https://www.w3.org/WAI/WCAG22/Understanding/keyboard.html) and [focus order](https://www.w3.org/WAI/WCAG22/Understanding/focus-order.html).
    **Proposed Solution:** Please see how EUI team solved that issue for **EuiIconTip**. 

2. **Clickable mode**: The tooltip button should have a unique `aria-label` attribute.  
   This violates the [WCAG 2.2 guideline on "Info and Relationships"](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html). 
Axe plugin trigger the following issue here: 

![Image](https://github.com/user-attachments/assets/a0fe7540-66cb-48e4-b826-05cea68c1400)

 **Proposed Solution:** We need some mechanizm for setting required `aria-label` attribute. 

Let's discuss how we can update the API to fix these issues.

**Comment by markov00** [comment]
@nickofthyme do you remember why this is a `button`? I don't think this is clickable, do we need that?

**Comment by nickofthyme** [comment]
Yeah it's clickable when there is a `Settings.onAnnotationClick` callback defined. It's also handles keyboard a11y but doesn't seem great and the rest of the chart has no keyboard interactions.

![Image](https://github.com/user-attachments/assets/13f48e6f-2941-4bb9-9d00-d52de054d158)

> This recording starts with an `onAnnotationClick` enabled where it's a `<button>` then I remove it and it reverts back to a `<div>`.
